src/test/java/uk/nhs/digital/nhsconnect/nhais/translator/AmendmentNameToEdifactTranslatorTest.java

```diff
@@ -0,0 +1,182 @@
+    private static final String FAMILY_NAME = "Smith";
+    private static final String FIRST_FORENAME = "John";
+    private static final String SECOND_FORENAME = "Adam";
+    private static final String OTHER_FORENAME = "Jacob";
+    private static final String TITLE = "Mr";
+
+    private final AmendmentNameToEdifactMapper translator = new AmendmentNameToEdifactMapper();
+
+    @Mock
+    private AmendmentBody amendmentBody;
+
+    @Mock
+    private JsonPatches jsonPatches;
+
+    @BeforeEach
+    void setUp() {
+        reset(amendmentBody, jsonPatches);
+
+        lenient().when(amendmentBody.getNhsNumber()).thenReturn(NHS_NUMBER);
+        lenient().when(jsonPatches.getAmendmentBody()).thenReturn(amendmentBody);
+        when(amendmentBody.getJsonPatches()).thenReturn(jsonPatches);
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = "getAddOrReplaceEnums")
+    void whenAddingOrReplacingAllFields_expectAllFieldsAreMapped(AmendmentPatchOperation operation) {
+        when(jsonPatches.getSurname()).thenReturn(Optional.of(new AmendmentPatch()
+            .setOp(operation).setValue(AmendmentValue.from(FAMILY_NAME))));
+        when(jsonPatches.getFirstForename()).thenReturn(Optional.of(new AmendmentPatch()
+            .setOp(operation).setValue(AmendmentValue.from(FIRST_FORENAME))));
+        when(jsonPatches.getSecondForename()).thenReturn(Optional.of(new AmendmentPatch()
+            .setOp(operation).setValue(AmendmentValue.from(SECOND_FORENAME))));
+        when(jsonPatches.getOtherForenames()).thenReturn(Optional.of(new AmendmentPatch()
+            .setOp(operation).setValue(AmendmentValue.from(OTHER_FORENAME))));
+        when(jsonPatches.getTitle()).thenReturn(Optional.of(new AmendmentPatch()
+            .setOp(operation).setValue(AmendmentValue.from(TITLE))));
+
+        var segments = translator.map(amendmentBody);
+
+        assertThat(segments).usingFieldByFieldElementComparator()
+            .containsExactly(PersonName.builder()
+                .nhsNumber(NHS_NUMBER)
+                .patientIdentificationType(PersonName.PatientIdentificationType.OFFICIAL_PATIENT_IDENTIFICATION)
+                .familyName(FAMILY_NAME)
+                .forename(FIRST_FORENAME)
+                .middleName(SECOND_FORENAME)
+                .thirdForename(OTHER_FORENAME)
+                .title(TITLE)
+                .build());
+    }
+
+    @Test
+    void whenRemovingAllFields_expectAllFieldsAreMapped() {
+        when(jsonPatches.getAllForenamesPath()).thenReturn(Optional.of(new AmendmentPatch()
+            .setOp(AmendmentPatchOperation.REMOVE)));
+        when(jsonPatches.getTitle()).thenReturn(Optional.of(new AmendmentPatch()
+            .setOp(AmendmentPatchOperation.REMOVE)));
+
+        var segments = translator.map(amendmentBody);
+
+        assertThat(segments).usingFieldByFieldElementComparator()
+            .containsExactly(PersonName.builder()
+                .nhsNumber(NHS_NUMBER)
```

Comment:
NHS Number is not a removable field

Reply:
Actually, its not "amendable" at all. Its also required as a parameter rather than a patch

Reply:
I'm not removing NHS number. I'm asserting that removing all fields produces a segment *with* NHS number and `%` in all other fields so it ends up in something like this `PNA+PAT+9999999999:OPI++++FO:%+TI:%'`